doc/compiler/memory/MemoryManager.md

```diff
@@ -15,90 +37,378 @@ by the Compiler; there are too many participants for informal trust,
+allocation that would cause the total allocated memory to exceed
+that limit will be rejected (i.e. `allocate()` will throw
+`std::bad_alloc()` ).
+
+The backing memory allocator has a diagnostic logging facility that
+uses macros like `MEMLOG`. Because the lowest level memory allocator
+layers are so fundamental in the compiler, one cannot trust that
+the higher level logging and options processing functions are
+available yet, so this facility relies only on `printf` and
+environment variables. The options availble for this facility
+are documented at the top of
+`compiler/env/mem/OMRBackingMemoryAllocator.cpp`. The level of
+detail can be set via the environment variable
+`OMRDebug_BackingMemory=<number>`. Turning on the debug facility
+will send trace output to `stdout` prefixed by the string `BckMem %p`
+where `%p` is the address of the `TR::BackingMemoryAllocator` object.
+
+#### `TR::SegmentAllocator`
+
+The segment allocator is the interface used by the higher level memory
+allocation functions. Like the backing memory allocator, the
+segment allocator returns memory wrapped in `TR::MemorySegment`s, but
+the segment allocator can allocate and will try to reuse smaller
+segment sizes. The backing memory allocator can be configured to
+allocate memory in 1MB segments, for example, but the segment allocator
+can then carve up those larger segments into smaller (64KB, say)
+segments to satisfy allocation requests made by the higher level
+memory allocation layer.
+
+So one key new feature of the segment allocator is that it can
+carve up the larger backing memory segments into smaller segments.
+The segment allocator maintains its own allocation block size
+(`_allocationBlockSize`).
+
+A second new feature is that the segment allocator will try to
+reuse segments that have been passed to `deallocate()` without
+returning them to the backing memory allocator. To avoid the
+overhead of OS level memory allocators, the segment allocator
+tries hard not to allocate more memory (with the consequence
+that it will hold onto allocated memory longer than strictly
+required by the compiler).
+
+This reuse has one restriction to simplify the reuse algorithm:
+only segments smaller than the default segment size (which will
+be rounded up to a multiple of the default segment size) can
+be reused. That means that larger "one-off" allocations will be
+satisfied by requesting a dedicated segment from the backing
+memory allocator, and such a segment will be immediately sent
+to the backing memory allocator to be deallocated when the
+compiler asks the segment allocator to deallocate it.
+
+To facilitate reuse, the segment allocator maintains a list of
+free segments, which are `TR::MemorySegment`s that have already
+been cut up into the default allocation size. Because all
+allocation requests are rounded up to this size anyway (except
+for larger memory requests), any segment on this free list can
```

Comment:
I find the phrasing of this sentence slightly confusing because, if I understand correctly, larger memory requests *are* rounded up to a *multiple* of the base size. Can the sentence be changed to something like:

```
Because all allocation requests smaller than the base are rounded up, ...
```

?

Reply:
i'll try to reword that phrase to make it less awkward